api/v1beta1/spannerautoscaler_webhook.go

```diff
@@ -38,9 +41,44 @@ var _ webhook.Defaulter = &SpannerAutoscaler{}
 
 // Default implements webhook.Defaulter so a webhook will be registered for the type
 func (r *SpannerAutoscaler) Default() {
-	spannerautoscalerlog.Info("default", "name", r.Name)
-
-	// TODO(user): fill in your defaulting logic.
+	log.Info("default", "name", r.Name)
+	log.V(1).Info("received spannerautoscaler resource for setting defaults", "name", r.Name, "resource", r)
+
+	// set the correct AuthType in Spec
+	switch {
+	case r.Spec.Authentication.IAMKeySecret != nil:
+		r.Spec.Authentication.Type = AuthTypeSA
+	case r.Spec.Authentication.ImpersonateConfig != nil:
+		r.Spec.Authentication.Type = AuthTypeImpersonation
+	default:
+		r.Spec.Authentication.Type = AuthTypeADC
+	}
+
+	// set appropriate namespace for secret
+	if r.Spec.Authentication.Type == AuthTypeSA && r.Spec.Authentication.IAMKeySecret.Namespace == "" {
+		r.Spec.Authentication.IAMKeySecret.Namespace = r.ObjectMeta.Namespace
+	}
+
+	// set default ComputeType
+	if r.Spec.ScaleConfig.ProcessingUnits != (ScaleConfigPUs{}) {
+		r.Spec.ScaleConfig.ComputeType = ComputeTypePU
+	}
+
+	// convert the nodes to processing units
+	if r.Spec.ScaleConfig.Nodes != (ScaleConfigNodes{}) {
+		r.Spec.ScaleConfig.ComputeType = ComputeTypeNode
+		r.Spec.ScaleConfig.ProcessingUnits = ScaleConfigPUs{
+			Min: r.Spec.ScaleConfig.Nodes.Min * 1000,
+			Max: r.Spec.ScaleConfig.Nodes.Max * 1000,
+		}
+	}
+
+	// set default ScaledownStepSize
+	if r.Spec.ScaleConfig.ScaledownStepSize == 0 {
+		r.Spec.ScaleConfig.ScaledownStepSize = 2
```

Comment:
https://github.com/mercari/spanner-autoscaler/pull/65/files#diff-81d59bd0d0ec7fde640490ad3785a60212a04c950c8b1afec25ecfae1608666fR94-R96

2000?

Reply:
Sorry, this has been fixed in #66, along with all other units also converted to PU. Let me know if that is acceptable - https://github.com/mercari/spanner-autoscaler/pull/66/files#diff-2291e5e50d322db09edb851bd8de9f2e93f60fc8e9ce64c508d8b46a5bb7921fR78

Reply:
I see, thanks!